transform_aes.c

```diff
@@ -141,28 +114,33 @@ static int transop_encode_aes( n2n_trans_op_t * arg,
             encode_uint32( outbuf, &idx, tx_sa_num ); // Not used
 
             /* Generate and encode the IV seed.
-             * Using two calls to rand() because RAND_MAX is usually < 64bit
-             * (e.g. linux) and sometimes < 32bit (e.g. Windows).
+             * Assume rand() to deliver 'RAND_BYTES' random bytes and re-calling rand() 
+             * because RAND_MAX is usually < 64bit (e.g. linux) 
+             * and sometimes < 32bit (e.g. Windows).
              */
-            iv_seed = ((((uint64_t)rand() & 0xFFFFFFFF)) << 32) | rand();
-            encode_buf(outbuf, &idx, &iv_seed, sizeof(iv_seed));
+	    uint32_t random_int;
+	    for (int i=0; i < TRANSOP_AES_IV_SEED_SIZE; i++) {
+		if ((i % RAND_BYTES) == 0) random_int = rand();
+		iv_seed[i] = (random_int >> ((i % RAND_BYTES) * 8)) & 0xFF;
+            }
+            encode_buf(outbuf, &idx, iv_seed, TRANSOP_AES_IV_SEED_SIZE);
 
             /* Encrypt the assembly contents and write the ciphertext after the SA. */
-            len = in_len + TRANSOP_AES_NONCE_SIZE;
+            len = in_len;
 
-            /* The assembly buffer is a source for encrypting data. The nonce is
-             * written in first followed by the packet payload. The whole
+            /* The assembly buffer is a source for encrypting data.
+             * Packet payload gets written in. The whole
              * contents of assembly are encrypted. */
-            pnonce = (uint32_t *)assembly;
-            *pnonce = rand();
-            memcpy( assembly + TRANSOP_AES_NONCE_SIZE, inbuf, in_len );
+            memcpy( assembly, inbuf, in_len );
 
             /* Need at least one encrypted byte at the end for the padding. */
             len2 = ( (len / AES_BLOCK_SIZE) + 1) * AES_BLOCK_SIZE; /* Round up to next whole AES adding at least one byte. */
             padding = (len2-len);
             assembly[len2 - 1] = padding;
-            traceEvent( TRACE_DEBUG, "padding = %u, seed = %016lx", padding, iv_seed );
 
+	    char * iv_seed_output[TRANSOP_AES_IV_SEED_SIZE*2];
```

Comment:
No star needed. Also, it should read `TRANSOP_AES_IV_SEED_SIZE*2+1` to take terminating \0 into account.